packages/ui-components/src/ApplePay/index.tsx

```diff
@@ -0,0 +1,193 @@
+    ApplePayClientMessages
+  >();
+
+  useLayoutEffect(() => {
+    if (initialized.current) return;
+    initialized.current = true;
+
+    const handleLoad = () => {
+      resize();
+      setSize({
+        width: button.current?.offsetWidth || 0,
+        height: button.current?.offsetHeight || 0,
+      });
+    };
+
+    const script = document.createElement("script");
+    script.src =
+      "https://applepay.cdn-apple.com/jsapi/1.latest/apple-pay-sdk.js";
+    script.async = true;
+    script.crossOrigin = "anonymous";
+    script.onload = handleLoad;
+    document.body.appendChild(script);
+  }, []);
+
+  const handleClick = async () => {
+    const session = buildSession(app, config);
+
+    session.oncancel = () => {
+      send("EV_APPLE_PAY_CANCELLED");
+    };
+
+    session.onpaymentauthorized = async (event) => {
+      try {
+        const {
+          payment: { token },
+        } = event;
+        const { paymentData } = token;
+
+        const encrypted: EncryptedApplePayData = await exchangeApplePaymentData(
+          app,
+          paymentData,
+          config.transaction.merchant.evervaultId
+        );
+
+        send("EV_APPLE_PAY_AUTH", encrypted);
+
+        const result =
+          await new Promise<ApplePayJS.ApplePayPaymentAuthorizationResult>(
+            (resolve) => {
+              on("EV_APPLE_PAY_COMPLETION", () => {
+                resolve({
+                  status: ApplePaySession.STATUS_SUCCESS,
+                });
+              });
+
+              on("EV_APPLE_PAY_AUTH_ERROR", (error) => {
+                if (error.code && error.contactField) {
+                  resolve({
+                    status: ApplePaySession.STATUS_FAILURE,
+                    errors: [
+                      {
+                        code: error.code,
+                        contactField: error.contactField,
+                        message: error.message,
+                      },
+                    ],
+                  });
+                } else {
+                  resolve({
+                    status: ApplePaySession.STATUS_FAILURE,
+                  });
+                }
+                const errorMsg = `Error during payment completion: ${error.message}`;
+                send("EV_APPLE_PAY_ERROR", errorMsg);
+              });
+            }
+          );
+
+        session.completePayment(result);
+      } catch (error) {
+        const errorMsg = `Error during payment completion. Error: ${error}`;
+        session.completePayment({
+          status: ApplePaySession.STATUS_FAILURE,
+        });
+        send("EV_APPLE_PAY_ERROR", errorMsg);
+      }
+    };
+
+    session.begin();
+  };
+
+  useEffect(() => {
+    const buttonElement = button.current;
+
+    if (buttonElement) {
+      buttonElement.addEventListener('click', handleClick);
+    }
+
+    return () => {
+      if (buttonElement) {
+        buttonElement.removeEventListener('click', handleClick);
+      }
+    };
+  }, []);
```

Comment:
The warning we are getting here is because the handleClick is defined outside the scope of the useEffect callback. We should probably move it inside before `const buttonElement = button.current` and also will  need to add `app` and `config` as dependencies of the useEffect array on line #154.